File: Backend/src/main/java/com/prueba/api/entity/FotosEjercicio.java
package com.prueba.api.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "fotosEjercicio")
public class FotosEjercicio {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    private String foto;

    @ManyToOne
    @JsonIgnoreProperties("fotosEjercicio")
    @JoinColumn(name = "ejercicio_id")
    private Ejercicio ejercicio;
}

File: Frontend/src/app/modelos/fotos-ejercicio.ts
import {Ejercicio} from "./ejercicio";

export class FotosEjercicio {
  id: string;
  foto: string;
  ejercicio: Ejercicio;

  constructor(id: string, foto: string, ejercicio: Ejercicio) {
    this.id = id;
    this.foto = foto;
    this.ejercicio = ejercicio;
  }
}
